core/src/autogluon/core/ray/manager.py

```diff
@@ -0,0 +1,20 @@
+from autogluon.core.utils.try_import import try_import_ray
+
+class RayManager:
+    """
+    A singleton maintaining status of ray cluster resources
+    """
+    
+    ray = None
+    current_config = None
+    
+    @staticmethod
+    def init_ray(**config):
+        """Init ray only if the configuration changes"""
+        if RayManager.ray is None:
+            RayManager.ray = try_import_ray()
+        if RayManager.current_config != config:
+            if RayManager.ray.is_initialized():
+                RayManager.ray.shutdown()
```

Comment:
Shutdown is a very time consuming process and we may avoid shutting down the cluster silently.

Reply:
Usually, the expectation is to call init once and shutdown once.

Reply:
We have to shutdown if we want to change the total resource

Reply:
A local shutdown is not too bad. In the case of a remote cluster, we are more than likely not needing this logic anymore

Reply:
If that’s the case, I recommend make it part of tabular rather than core. Functions in core should be concise and reusable.

Reply:
So this is not tabular only. All this is doing is if the user wants to init with a different config, do it. Otherwise, it will do nothing. Why it's not reusable?